torch/_export/utils.py

```diff
@@ -13,20 +13,33 @@
     FlattenFunc,
     FromDumpableContextFn,
     ToDumpableContextFn,
-    tree_flatten,
     UnflattenFunc,
+    tree_flatten,
 )
+import torch.fx._pytree as fx_pytree
 
 
 SERIALIZED_DATACLASS_TO_PYTHON_DATACLASS: Dict[str, Type[Any]] = {}
 
 
 @torch._dynamo.disable
 def _check_input_constraints_pre_hook(self, *args, **kwargs):
-    flat_args, _ = tree_flatten(args)
+    try:
+        args = fx_pytree.tree_flatten_spec(
```

Comment:
avoid tree_flatten_spec. It's a weird forked thing and doesn't support a bunch of features, notably including custom pytrees.